exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/BasicFormatMatcher.java

```diff
@@ -72,7 +72,7 @@ public boolean supportDirectoryReads() {
 
   @Override
   public FormatSelection isReadable(DrillFileSystem fs, FileSelection selection) throws IOException {
-    if (isReadable(fs, selection.getFirstPath(fs))) {
+    if (isFileReadable(fs, selection.getFirstPath(fs))) {
```

Comment:
You are right, ParquetFormatMatcher overrides isReadable to first check if the directory contains the metadata file, if not it delegates to the base class method to check if a particular file is readable or not. 

The existing naming is a bit confusing. In BasicFormatMatcher there are two methods with the same name:

1. FormatSelection isReadable(DrillFileSystem fs, FileSelection selection) -> operates on directory level
2. boolean isReadable(DrillFileSystem fs, FileStatus status) -> operates on a single file level.

I've renamed the second to be called isFileReadable and exposed it in the abstract class so drop can invoke it on every file.